MelonLoader.Bootstrap/PltHook.cs

```diff
@@ -0,0 +1,61 @@
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using MelonLoader.Bootstrap.Logging;
+
+namespace MelonLoader.Bootstrap;
+
+internal static partial class PltHook
+{
+    [LibraryImport("*", EntryPoint = "plthook_open", StringMarshalling = StringMarshalling.Utf8)]
+    [UnmanagedCallConv(CallConvs = [typeof(CallConvCdecl)])]
+    private static partial int PlthookOpen(ref nint pltHookOut, string? filename);
+
+    [LibraryImport("*", EntryPoint = "plthook_open_by_handle")]
+    [UnmanagedCallConv(CallConvs = [typeof(CallConvCdecl)])]
+    private static partial int PlthookOpenByHandle(nint pltHookOut, nint handle);
+
+    [LibraryImport("*", EntryPoint = "plthook_open_by_address")]
+    [UnmanagedCallConv(CallConvs = [typeof(CallConvCdecl)])]
+    private static partial int PlthookOpenByAddress(nint pltHookOut, nint address);
+
+    [LibraryImport("*", EntryPoint = "plthook_replace", StringMarshalling = StringMarshalling.Utf8)]
+    [UnmanagedCallConv(CallConvs = [typeof(CallConvCdecl)])]
+    private static unsafe partial int PlthookReplace(nint pltHook, string funcName, nint funcAddr, nint oldFunc);
+
+    [LibraryImport("*", EntryPoint = "plthook_close")]
+    [UnmanagedCallConv(CallConvs = [typeof(CallConvCdecl)])]
+    private static partial void PlthookClose(nint pltHook);
+
+    [LibraryImport("*", EntryPoint = "plthook_error")]
+    [UnmanagedCallConv(CallConvs = [typeof(CallConvCdecl)])]
+    private static partial nint PlthookError();
+
+    private static readonly string? LibraryFileName = Process.GetCurrentProcess().Modules.OfType<ProcessModule>()
+        .FirstOrDefault(x => x.FileName.Contains("UnityPlayer"))?.FileName;
```

Comment:
Since this class is meant to be a wrapper for PltHook, it shouldn't be specific to any library. This does not belong here

Reply:
Considering this is somewhat of an alternative to Dobby, it should mirror the API. This would allow the caller to maintain the hooks properly.

Also, there is no point in accepting multiple hooks at the same time. Opening and closing PltHook multiple times shouldn't cause issues.